File: apis/elysium-character-api/src/main/java/com/seventh_root/elysium/api/character/RaceProvider.java
package com.seventh_root.elysium.api.character;

import com.seventh_root.elysium.core.service.ServiceProvider;

import java.util.Collection;

public interface RaceProvider<T extends Race> extends ServiceProvider {

    T getRace(int id);

    T getRace(String name);

    Collection<? extends T> getRaces();

    void addRace(T race);

    void removeRace(T race);

}

File: apis/elysium-character-api/src/main/kotlin/com/seventh_root/elysium/api/character/RaceProvider.kt
package com.seventh_root.elysium.api.character

import com.seventh_root.elysium.core.service.ServiceProvider

interface RaceProvider<T : Race> : ServiceProvider {

    fun getRace(id: Int): Race?
    fun getRace(name: String): Race?
    val races: Collection<T>
    fun addRace(race: T)
    fun removeRace(race: T)

}
